scripts/Diffraction/wish/wishSX.py

```diff
@@ -0,0 +1,129 @@
+                if "<sphere" in ADS.retrieve(wsname).sample().getShape().getShapeXML():
+                    transmission = mantid.SphericalAbsorption(InputWorkspace=wsname, OutputWorkspace="transmission", EnableLogging=False)
+                else:
+                    transmission = mantid.MonteCarloAbsorption(
+                        InputWorkspace=wsname, OutputWorkspace="transmission", EventsPerPoint=self.n_mcevents, EnableLogging=False
+                    )
+                self._divide_workspaces(wsname, transmission)
+                mantid.DeleteWorkspace(transmission)
+            # save results in dictionary
+            self.set_ws(run, wsname)
+        return wsname
+
+    @staticmethod
+    def mask_detector_edges(ws, nedge=16, ntubes=2):
+        # mask pixels on ends of tubes
+        mantid.MaskBTP(Workspace=ws, Pixel=f"1-{nedge},{512-nedge}-512")
+        # only mask tubes on panels with edge facing beam in/out
+        tubes = np.r_[152 - (ntubes - 1) // 2 : 153, 76 - ntubes // 2 + 1 : 77]
+        mantid.MaskBTP(Workspace=ws, Bank="5-6", Tube=",".join(str(tube) for tube in tubes))
+        tubes = np.r_[1 : (ntubes + 1) // 2 + 1, 77 : 77 + ntubes // 2]
+        mantid.MaskBTP(Workspace=ws, Bank="1,10", Tube=",".join(str(tube) for tube in tubes))
+
+    @BaseSX.default_apply_to_all_runs
+    def convert_to_MD(self, run=None, frame="Q (lab frame)"):
+        wsname = WishSX.retrieve(self.get_ws(run)).name()
+        WishSX.mask_detector_edges(wsname, nedge=16, ntubes=2)
+        super().convert_to_MD(run, frame)
+
+    def process_vanadium(self):
+        # vanadium
+        self.van_ws = self.load_run(self.van_runno)
+        mantid.SmoothNeighbours(InputWorkspace=self.van_ws, OutputWorkspace=self.van_ws, Radius=3)
+        mantid.SmoothData(InputWorkspace=self.van_ws, OutputWorkspace=self.van_ws, NPoints=301)
+        # correct vanadium for absorption
+        mantid.ConvertUnits(InputWorkspace=self.van_ws, OutputWorkspace=self.van_ws, Target="Wavelength", EnableLogging=False)
+        mantid.SetSample(
+            self.van_ws,
+            Geometry={"Shape": "CSG", "Value": self.sphere_shape},
+            Material={"ChemicalFormula": "V0.95-Nb0.05", "SampleNumberDensity": 0.0722},
+        )
+        vanadium_transmission = mantid.SphericalAbsorption(InputWorkspace=self.van_ws, OutputWorkspace="vanadium_transmission")
+        self._divide_workspaces(self.van_ws, vanadium_transmission)
+        mantid.DeleteWorkspace(vanadium_transmission)
+
+    @staticmethod
+    def load_run(runno):
+        wsname = "WISH000" + str(runno)
+        ws, mon = mantid.LoadRaw(Filename=wsname + ".raw", OutputWorkspace=wsname, LoadMonitors="Separate")
+        mon_name = mon.name()
+        mantid.CropWorkspace(InputWorkspace=wsname, OutputWorkspace=wsname, XMin=6000, XMax=99000)
+        mantid.CropWorkspace(InputWorkspace=mon_name, OutputWorkspace=mon_name, XMin=6000, XMax=99000)
```

Comment:
Similar comment to sxd.py. I think you should pull out these crop ranges as constants